### ISSUE TITLE
creating aarch64 windows VM fails: virt-manager shouldn't try to set hyperv bits for non-x86

### ISSUE THREAD

**Issue by lacamar** [virt-manager/virt-manager#509]
I'm attempting to run a Windows 11 arm ISO (from UUP dump) on Asahi Linux (M1 Pro CPU), but I receive this error:

```
Unable to complete install: 'internal error: process exited while connecting to monitor: 2023-04-20T12:55:10.052478Z qemu-system-aarch64: can't apply global host-arm-cpu.hv-relaxed=on: Property 'host-arm-cpu.hv-relaxed' not found'

Traceback (most recent call last):
  File "/usr/share/virt-manager/virtManager/asyncjob.py", line 71, in cb_wrapper
    callback(asyncjob, *args, **kwargs)
  File "/usr/share/virt-manager/virtManager/createvm.py", line 2008, in _do_async_install
    installer.start_install(guest, meter=meter)
  File "/usr/share/virt-manager/virtinst/install/installer.py", line 695, in start_install
    domain = self._create_guest(
  File "/usr/share/virt-manager/virtinst/install/installer.py", line 637, in _create_guest
    domain = self.conn.createXML(initial_xml or final_xml, 0)
  File "/usr/lib/python3.10/site-packages/libvirt.py", line 4481, in createXML
    raise libvirtError('virDomainCreateXML() failed')
libvirt.libvirtError: internal error: process exited while connecting to monitor: 2023-04-20T12:55:10.052478Z qemu-system-aarch64: can't apply global host-arm-cpu.hv-relaxed=on: Property 'host-arm-cpu.hv-relaxed' not found
```

When I try to set up the VM as Generic or Unknown OS, I receive no error from virt-manager but when the VM runs, this is all that appears on the terminal:

```
BdsDxe: loading Boot0001 "UEFI QEMU QEMU CD-ROM " from PciRoot(0x0)/Pci(0x1,0x2)/Pci(0x0,0x0)/Scsi(0x0,0x0)
BdsDxe: starting Boot0001 "UEFI QEMU QEMU CD-ROM " from PciRoot(0x0)/Pci(0x1,0x2)/Pci(0x0,0x0)/Scsi(0x0,0x0)
```

I installed Virt-manager via the AUR as virt-manager-git and I built QEMU 8.0.0 from source.

**Comment by rokejulianlockhart** [comment]
How did you capture those logs, @lacamar?

**Comment by lacamar** [comment]
It was from the last step of the interactive VM creation process in virt-manager. It creates an error window with that output.

**Comment by crobinso** [comment]
Thanks for the report! Sorry for the late response.

I can reproduce, trying to create an aarch64 windows VM. It's because of some of the hyperv XML bits we are trying to set in the XML, apparently at the qemu level they are x86 only. So this is a virt-manager bug.

**Comment by lacamar** [comment]
All good, I'm just glad to see that this may be possible to resolve since I'm still interested in virtualising windows on aarch64. Let me know if I need to do any further testing on my end.

**Comment by crobinso** [comment]
@lacamar this should be fixed in git now, the VM config will hopefully boot. Would be interested to see if at least the VM launches now.

I'm going to close this issue. If you encounter further problems, please open a new issue and provide full `virt-manager --debug` output when reproducing, and I will try to help from there.

**Comment by RadxaYuntian** [comment]
Hope we get commit to track the fix in the future. I'm currently encountering this issue on Debian 12 while it was working on NixOS. No idea if I can just patch some python scripts or if I need to recompile libvirt right now.

**Comment by gmarkall** [comment]
The commits are:

* c279c17e6c62a8395f30c6ae04ee010019881598
* 66bbfa23b228c773906b8051844b3400ced377fa
* 9e9dbf735ca558b9b2dc9583c912cd083fd22029

I was able to workaround the issue by applying these commits on top of an installed virt-manager 4.1.0, which seems to be the version included with Ubuntu 24.04.
